sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1208,36 +1204,72 @@ object SQLContext {
   private val INSTANTIATION_LOCK = new Object()
 
   /**
-   * Reference to the last created SQLContext.
+   * The active SQLContext for threads.
+   */
+  private val activeContexts: InheritableThreadLocal[SQLContext] =
+    new InheritableThreadLocal[SQLContext]
+
+  /**
+   * Reference to the created SQLContext.
    */
-  @transient private val lastInstantiatedContext = new AtomicReference[SQLContext]()
+  @transient private val instantiatedContext = new AtomicReference[SQLContext]()
```

Comment:
I think this makes more sense than "last" but we should update the objects doc if we are going to change the semantics.